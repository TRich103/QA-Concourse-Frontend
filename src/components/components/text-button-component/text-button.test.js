import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme,{shallow, mount, render, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import axios from "axios";
import MockAdapter from 'axios-mock-adapter';
import renderer from 'react-test-renderer';
import TextButton from './text-button.component';
import { Button, ButtonGroup } from 'reactstrap';

configure({ adapter: new Adapter() });

const clickFn = jest.fn();

describe('TextButton', () => {

  it('should be disabled when disabled is set to true', () => {
    const boolean = true;
    
    const component = shallow(<TextButton disabled={boolean} />);
  
    expect(component).toMatchSnapshot();
  });

  it('should render button text correctly with given string', () => {
    const string = 'Click Me';
    
    const component = shallow(<TextButton text={string}/>);
  
    expect(component).toMatchSnapshot();
  });

  it('button click should call function', () => {
    const string = 'my-button';
      const component = mount(<TextButton id={string} onClick={clickFn}  />);
      
      component
        .find('button#my-button')
        .simulate('click');
      expect(clickFn).toHaveBeenCalled();
      component.unmount(); 
    });
});

