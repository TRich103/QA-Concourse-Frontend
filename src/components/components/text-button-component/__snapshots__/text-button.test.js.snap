// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextButton should be disabled when disabled is set to true 1`] = `
<div>
  <ButtonGroup
    role="group"
    tag="div"
  >
    <Button
      color="secondary"
      disabled={true}
      tag="button"
    />
  </ButtonGroup>
</div>
`;

exports[`TextButton should render button text correctly with given string 1`] = `
<div>
  <ButtonGroup
    role="group"
    tag="div"
  >
    <Button
      color="secondary"
      tag="button"
    >
      Click Me
    </Button>
  </ButtonGroup>
</div>
`;
